packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,244 @@
+        items: void 0,
+        managedClasses: void 0,
+    };
+
+    private tabsRef: React.RefObject<BaseTabs>;
+
+    private ltr: Direction;
+
+    private prevTabPanelIndex: number;
+
+    /**
+     * The constructor
+     */
+    constructor(props: PivotProps) {
+        super(props);
+
+        if (Array.isArray(this.props.items)) {
+            this.state = {
+                offsetX: 0,
+                tabPanelIndex: 0,
+                activeId:
+                    typeof this.props.activeId === "string"
+                        ? this.props.activeId
+                        : get(this.props.items[0], "id", ""),
+            };
+        }
+
+        this.tabsRef = React.createRef();
+    }
+
+    public componentDidMount(): void {
+        this.ltr = this.getLTR();
+        this.setActiveIndicatorOffset();
+        this.prevTabPanelIndex = this.state.tabPanelIndex;
+    }
+
+    public componentDidUpdate(prevProps: PivotProps, prevState: PivotState): void {
+        if (this.ltr !== this.getLTR()) {
+            this.setActiveIndicatorOffset();
+            this.ltr = this.getLTR();
+        }
+
+        if (this.state.activeId !== prevState.activeId) {
+            this.setActiveIndicatorOffset();
+            this.updateTabPanelIndex();
+        }
+
+        if (this.state.tabPanelIndex !== prevState.tabPanelIndex) {
+            this.prevTabPanelIndex = this.state.tabPanelIndex;
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <BaseTabs
+                {...this.unhandledProps()}
+                ref={this.tabsRef}
+                managedClasses={this.generatePivotClassNames()}
+                activeId={this.state.activeId}
+                onUpdate={this.handleTabsUpdate}
+                items={this.props.items}
+                label={this.props.label}
+            >
+                <span
+                    style={{ transform: `translateX(${toPx(this.state.offsetX)})` }}
+                    className={get(
+                        this.props,
+                        "managedClasses.pivot_activeIndicator",
+                        ""
+                    )}
+                />
+            </BaseTabs>
+        );
+    }
+
+    /**
+     * Returns tabs managedclasses with new carousel-specific JSS
+     */
+    protected generatePivotClassNames(): TabsClassNameContract {
+        return {
+            tabs: get(this.props, "managedClasses.pivot", ""),
+            tabs_tabPanels: this.generateTabPanelsClassNames(),
+            tabs_tabList: get(this.props, "managedClasses.pivot_tabList", ""),
+            tabs_tabContent: get(this.props, "managedClasses.pivot_tabContent", ""),
+            tabs_tabPanelContent: get(
+                this.props,
+                "managedClasses.pivot_tabPanelContent",
+                ""
+            ),
+            tab: get(this.props, "managedClasses.pivot_tab", ""),

```

Comment:
have you tested what these will do if you only defined some of them? (eg. will they result in "undefined" in the class attribute that is rendered)